src/utarray.h

```diff
@@ -217,6 +217,17 @@ typedef struct {
   qsort((a)->d, (a)->i, (a)->icd.sz, cmp);                                    \
 } while(0)
 
+#define utarray_replace(a, j, ele) do {                                           \
+  char *ptr = utarray_eltptr(a, j);                                               \
+  if (ptr) {                                                                      \
```

Comment:
Why is this test here? Your (very short) documentation doesn't mention anything about only _conditionally_ replacing the element. If you want the semantics to be "Replace the given index, _unless_ that index is out of range, in which case this function is a no-op," then that definitely needs to be in your documentation, because it's not obvious at all.

At first glance I suggest removing this test: `j` should always refer to a valid element, so `ptr` should never be null, so the test can be removed. (That is, just like `utarray_erase`, this function has a _precondition_ that the inputs are valid.)